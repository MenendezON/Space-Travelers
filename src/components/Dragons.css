.dragon {
  width: 1080px;
  margin: 50px auto 0;
  display: flex;
  gap: 10px;
  flex-wrap: wrap;
  justify-content: space-between;
}

.dragon .card {
  background-color: white;
  width: 500px;
  border: 2px solid silver;
  border-radius: 8px;
  padding: 5px;
}

.mainImg {
  display: flex;
  justify-content: end;
  margin: 5px 0;
}

.button,
.reserved {
  height: 25px;
  border: 0;
  background-color: gray;
  font-weight: bold;
  cursor: pointer;
  color: white;
}

.button {
  background-color: #f66f1c;
}

.reserved {
  background-color: gray;
}

.button:hover,
.reserved:hover {
  color: silver;
}

.images > img {
  height: 90px;
  cursor: pointer;
  border: 2px solid white;
}

.dragon .card > img {
  height: 300px;
  margin-top: 5px;
  background-color: rgba(255, 255, 255, 0.4);
}

.dragon .card h2,
.dragon .card em {
  margin: 0;
  padding: 5px;
}

.dragon .card h2 {
  color: gray;
}

.images {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.images > img:hover {
  border: 2px solid black;
}
