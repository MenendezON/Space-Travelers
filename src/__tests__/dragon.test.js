import React from 'react';
import { render } from '@testing-library/react';
import { useSelector } from 'react-redux';
import Dragons from '../components/dragons';

// Mocking react-redux's useSelector
jest.mock('react-redux');

// Mocking axios
jest.mock('axios', () => ({
  get: jest.fn(),
}));

test('Missions component renders correctly', () => {
  // Mock the useSelector behavior
  useSelector.mockReturnValue([
    {
      id: 1, name: 'Dragon 1', type: 'Type 1', flickr_images: ['image1.jpg'], reserved: false,
    },
    {
      id: 2, name: 'Dragon 2', type: 'Type 2', flickr_images: ['image2.jpg'], reserved: true,
    },
    // Add more mission objects as needed
  ]);

  const { container } = render(<Dragons />);

  expect(container).toMatchSnapshot();
});
