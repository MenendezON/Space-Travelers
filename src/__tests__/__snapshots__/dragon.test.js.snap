// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions component renders correctly 1`] = `
<div>
  <section
    class="dragon"
  >
    <div
      class="card"
    >
      <div>
        <h2>
          Dragon 1
        </h2>
        <em>
          Type 1
        </em>
      </div>
      <div
        class="mainImg"
        style="height: 300px; background-image: url(image1.jpg); background-size: cover;"
      >
        <button
          class="button"
          type="button"
        >
          Reserve dragon
        </button>
      </div>
      <div
        class="images"
      >
        <img
          alt=""
          src="image1.jpg"
        />
      </div>
    </div>
    <div
      class="card"
    >
      <div>
        <h2>
          Dragon 2
        </h2>
        <em>
          Type 2
        </em>
      </div>
      <div
        class="mainImg"
        style="height: 300px; background-image: url(image2.jpg); background-size: cover;"
      >
        <button
          class="reserved"
          type="button"
        >
          Cancel the reserve
        </button>
      </div>
      <div
        class="images"
      >
        <img
          alt=""
          src="image2.jpg"
        />
      </div>
    </div>
  </section>
</div>
`;
